### DISCUSSION TITLE
Date received in the WebHook

### DISCUSSION THREAD

**Discussion by daniel89fg** [netflie/whatsapp-cloud-api#91] [Q&A]
Hello.

I have a question that I can't understand, even reading the documentation. Please I would like to know if you can clarify it for me.

When a user sends a message, it reaches Whatsapp and sends it to us via POST, and we receive it with the WebHook class.

The official documentation says that there are 3 dates: delivered, read and sent.

When we receive the message and want to save it in our database, we recover the timestamp field of the message with the receivedAt() function, correct?

But it is not clear to me if that date belongs to "sent" or to "delivered", or to which one.

Could you clarify this doubt for me?

Another thing I see is that the date received with receivedAt() is a class of DateTimeImmutable, where we also have the time zone. I understand it is the time zone from which the message was sent, correct?

Thank you very much, greetings.

**Comment by aalbarca** [comment]
Read, delivered and sent aren't dates. They are events.

Those events are received when **your customer received or read** a message.

Best regards.
